misc/python/materialize/cli/ci_annotate_errors.py

```diff
@@ -233,44 +266,76 @@ def annotate_logged_errors(log_files: list[str]) -> int:
                     break
             else:
                 unknown_errors.append(
-                    f"Unknown error in {linked_file}:  \n{formatted_error_message}"
+                    f"Unknown error in {location}:\n{formatted_error_message}"
                 )
 
-    annotate_errors(
-        unknown_errors,
-        "Unknown errors and regressions in logs (see [ci-regexp](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/ci-regexp.md))",
-        "error",
-    )
-    annotate_errors(known_errors, "Known errors in logs, ignoring", "info")
+    for error in errors:
+        if isinstance(error, ErrorLog):
+            for artifact in artifacts:
+                if artifact["job_id"] == job and artifact["path"] == error.file:
+                    org = os.environ["BUILDKITE_ORGANIZATION_SLUG"]
+                    pipeline = os.environ["BUILDKITE_PIPELINE_SLUG"]
+                    build = os.environ["BUILDKITE_BUILD_NUMBER"]
+                    linked_file: str = f'[{error.file}](https://buildkite.com/organizations/{org}/pipelines/{pipeline}/builds/{build}/jobs/{artifact["job_id"]}/artifacts/{artifact["id"]})'
+                    break
+            else:
+                linked_file: str = error.file
+
+            handle_error(error.match, linked_file)
+
+        else:
+            if "in Code Coverage" in error.text:
+                # Don't bother looking up known issues for code coverage report, just print it verbatim as an info message
+                known_errors.append(f"{error.testcase}:\n{error.text}")
+            else:
+                handle_error(error.text.encode("utf-8"), error.testcase)
+
+    annotate_errors(unknown_errors, known_errors, get_failures_on_main())
 
     if unknown_errors:
-        print(
-            f"+++ Failing test because of {len(unknown_errors)} unknown error(s) in logs:"
-        )
+        print(f"+++ Failing test because of {len(unknown_errors)} unknown error(s):")
         print(unknown_errors)
 
     return len(unknown_errors)
 
 
-def get_error_logs(log_file_names: list[str]) -> list[ErrorLog]:
+def get_errors(log_file_names: list[str]) -> list[ErrorLog | JunitError]:
     error_logs = []
     for log_file_name in log_file_names:
-        with open(log_file_name, "r+") as f:
-            try:
-                data: Any = mmap.mmap(f.fileno(), 0)
-            except ValueError:
-                # empty file, ignore
-                continue
+        if "junit_" in log_file_name:
+            xml = JUnitXml.fromfile(log_file_name)
+            for suite in xml:
+                for case in suite:
+                    for result in case.result:
+                        if not isinstance(result, Error) and not isinstance(
+                            result, Failure
+                        ):
+                            continue
+                        # junit_testdrive_* is excluded by this, but currently
```

Comment:
Do we generate two files for testdrive at the moment?

Reply:
Yes! One seems pretty useless.